Source/VungleAdapterBannerAd.swift

```diff
@@ -5,268 +5,146 @@
 
 import ChartboostMediationSDK
 import Foundation
-import VungleSDK
+import VungleAdsSDK
 
 /// Chartboost Mediation Vungle adapter banner ad.
 final class VungleAdapterBannerAd: VungleAdapterAd, PartnerAd {
-    
+
+    var ad: VungleBanner?
+
     /// The partner ad view to display inline. E.g. a banner view.
     /// Should be nil for full-screen ads.
     var inlineView: UIView? = UIView()
-    
+
     /// Indicates if the Vungle banner was displayed with a successful call to Vungle SDK's `addAdView`.
     private var adWasDisplayed = false
-    
+
     /// Loads an ad.
     /// - parameter viewController: The view controller on which the ad will be presented on. Needed on load for some banners.
     /// - parameter completion: Closure to be performed once the ad has been loaded.
     func load(with viewController: UIViewController?, completion: @escaping (Result<PartnerEventDetails, Error>) -> Void) {
         log(.loadStarted)
-        
-        // If ad already loaded then attach it and finish immediately
-        guard !adIsCachedByVungle else {
-            do {
-                // Attach vungle view to the container
-                try addVungleAdToContainer()
-                
-                // Report load success
-                log(.loadSucceeded)
-                completion(.success([:]))
-            } catch {
-                // Report load failure
-                log(.loadFailed(error))
-                completion(.failure(error))
-            }
-            return
-        }
-        
         loadCompletion = completion
-        
-        // If ad loading already in progress wait for it to finish.
-        // Vungle does not handle well loadPlacement() calls when a load for the same placement is already ongoing.
-        // This may happen after a PartnerAd has been created and invalidated, since Vungle will keep the load going
-        // even after Chartboost Mediation has discarded the PartnerAd instance.
-        if router.isLoadInProgress(for: request) {
-            // `loadCompletion` is executed when the ongoing load finishes leading to a `vungleAdPlayabilityUpdate()` call
-            return
-        }
-        
-        // Start loading
-        router.recordLoadStart(for: request)
-        do {
-            try loadVungleAd()
-        } catch {
-            router.recordLoadEnd(for: request)
-            log(.loadFailed(error))
-            completion(.failure(error))
-            loadCompletion = nil
-        }
+
+        ad = VungleBanner(placementId: request.partnerPlacement, size: self.vungleAdSize)
+        ad?.delegate = self
+        // If the adm is nil, that's the same as telling it to load a non-programatic ad
+        ad?.load(request.adm)
```

Comment:
Reading this code it's not clear if the VungleBanner init returns an optional or a non-optional object.
Can we create a local variable to hold the non-optional banner, interact with it, and later assign it to the optional instance property?
If the init return is actually an optional, can we unwrap it and fail the load if it's nil?

Reply:
[Done](https://github.com/ChartBoost/chartboost-mediation-ios-adapter-vungle/pull/44/commits/c521b30e6bc2458d4cab31fe06bdcaca619962c0). Also simplified `bannerAdDidLoad` by using the non-optional banner reference that's passed into the delegate method instead of the reference stored in `self`.
(But if I don't hold onto a reference on `self`, the delegate never gets called - I think because the ad goes out of scope and gets garbage collected after `load()`)